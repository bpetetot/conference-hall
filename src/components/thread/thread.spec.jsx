/* eslint-env jest */
import React from 'react'
import { mount } from 'enzyme'
import Thread from './index'

describe('components/thread', () => {
  it('should render a message in the thread', () => {
    const messages = [
      {
        id: 'message1',
        name: 'Luke Skywalker',
        message: 'May the force be with you',
        date: new Date(0),
        img: 'https://upload.wikimedia.org/wikipedia/en/9/9b/Luke_Skywalker.png',
        onSave: () => {},
        allowEdit: false,
      },
    ]
    const mock = jest.fn()
    const wrapper = mount(<Thread
      currentUser="Yoda"
      messages={messages}
      onSaveMessage={mock}
      onDeleteMessage={mock}
    />)
    expect(wrapper).toMatchSnapshot()
  })
})
