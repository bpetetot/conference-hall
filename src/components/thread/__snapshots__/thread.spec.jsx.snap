// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/thread should render a message in the thread 1`] = `
<Thread
  currentUser="Yoda"
  messages={
    Array [
      Object {
        "allowEdit": false,
        "date": 1970-01-01T00:00:00.000Z,
        "id": "message1",
        "img": "https://upload.wikimedia.org/wikipedia/en/9/9b/Luke_Skywalker.png",
        "message": "May the force be with you",
        "name": "Luke Skywalker",
        "onSave": [Function],
      },
    ]
  }
  onDeleteMessage={[MockFunction]}
  onSaveMessage={[MockFunction]}
>
  <div
    className="thread"
  >
    <div
      className="messages"
    >
      <Message
        allowEdit={false}
        className="message"
        date={1970-01-01T00:00:00.000Z}
        id="message1"
        img="https://upload.wikimedia.org/wikipedia/en/9/9b/Luke_Skywalker.png"
        key="0"
        message="May the force be with you"
        modified={false}
        name="Luke Skywalker"
        onDelete={[MockFunction]}
        onSave={[MockFunction]}
      >
        <div
          className="wrapper message"
        >
          <Avatar
            className="avatar"
            name="Luke Skywalker"
            size="medium"
            square={false}
            src="https://upload.wikimedia.org/wikipedia/en/9/9b/Luke_Skywalker.png"
            withLabel={false}
          >
            <div
              className="cc-avatar-wrapper avatar"
            >
              <div
                className="cc-avatar cc-avatar-image"
                style={
                  Object {
                    "background": "transparent",
                  }
                }
              >
                <img
                  alt="Luke Skywalker"
                  src="https://upload.wikimedia.org/wikipedia/en/9/9b/Luke_Skywalker.png"
                />
              </div>
            </div>
          </Avatar>
          <div
            className="messageContent"
          >
            <div
              className="message"
            >
              <span
                className="name"
              >
                Luke Skywalker
              </span>
              <span
                className="date"
              >
                over 49 years ago
              </span>
              <span
                className="modified"
              />
            </div>
            <div
              className="message"
            >
              May the force be with you
            </div>
          </div>
        </div>
      </Message>
    </div>
    <div
      className="input"
    >
      <input
        name="message"
        onKeyDown={[Function]}
        placeholder="Send a message"
        type="text"
      />
      <Button
        accent={false}
        block={false}
        error={false}
        onClick={[Function]}
        primary={true}
        secondary={false}
        simple={false}
        size="normal"
        tertiary={false}
      >
        <button
          className="cc-button cc-button-primary"
          onClick={[Function]}
          type="button"
        >
          Send
        </button>
      </Button>
    </div>
  </div>
</Thread>
`;
